---
title: "scrna-data-metrics"
author: "Genevieve Housman"
date: "December 2022"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(tidyverse)
library(Seurat)
```


# scRNA-seq Data Metrics

View 10X data metrics using seurat objects

Adjust cellplex tag assignment thresholds

```{r}
cp_threshold_assignment <- function(data, threshold){
  data %>% 
    select(Row, Barcode) %>% 
    full_join(  
      data %>% 
      pivot_longer(
        cols = c(contains("CMO"), Multiplet, Blank),
        names_to = "Tag",
        values_to = "Probability"
      ) %>% 
      filter(Probability > threshold) %>% 
      #filter(!Tag %in% c("Blank", "Multiplet")) %>% 
      group_by(Barcode) %>% 
      #summarise(Assigned_Tags = paste(Tag, collapse = "|")),
      summarise(Assigned_Tags = paste(Tag, collapse = "|"), Assigned_Probability = Probability),
      by = "Barcode"
    ) %>% 
    replace_na(list(Assigned_Tags = "Unassigned"))
}

#batch 1 using human reference
x <- read.csv("../scRNA/human_chimp_chondro_time_batch1_h/outs/multi/multiplexing_analysis/assignment_confidence_table.csv")
x_2 <- cp_threshold_assignment(data=x, threshold=0.5)
barcode_assign_b1 <- merge(x[,c("Barcode","Assignment","Assignment_Probability")], x_2[,c(2:4)], by="Barcode")
colnames(barcode_assign_b1)[2:5] <- c("HumanRef_0.9","HumanRef_0.9_AssignProb","HumanRef_0.5","HumanRef_0.5_AssignProb")

#batch 1 using chimp reference
x <- read.csv("../scRNA/human_chimp_chondro_time_batch1_c/outs/multi/multiplexing_analysis/assignment_confidence_table.csv")
x_2 <- cp_threshold_assignment(data=x, threshold=0.5)
barcode_assign_b1 <- merge(barcode_assign_b1, x[,c("Barcode","Assignment","Assignment_Probability")], by="Barcode")
barcode_assign_b1 <- merge(barcode_assign_b1, x_2[,c(2:4)], by="Barcode")
colnames(barcode_assign_b1)[6:9] <- c("ChimpRef_0.9","ChimpRef_0.9_AssignProb","ChimpRef_0.5","ChimpRef_0.5_AssignProb")

#batch 2 using human reference
x <- read.csv("../scRNA/human_chimp_chondro_time_batch2_h/outs/multi/multiplexing_analysis/assignment_confidence_table.csv")
x_2 <- cp_threshold_assignment(data=x, threshold=0.5)
barcode_assign_b2 <- merge(x[,c("Barcode","Assignment","Assignment_Probability")], x_2[,c(2:4)], by="Barcode")
colnames(barcode_assign_b2)[2:5] <- c("HumanRef_0.9","HumanRef_0.9_AssignProb","HumanRef_0.5","HumanRef_0.5_AssignProb")

#batch 2 using chimp reference
x <- read.csv("../scRNA/human_chimp_chondro_time_batch2_c/outs/multi/multiplexing_analysis/assignment_confidence_table.csv")
x_2 <- cp_threshold_assignment(data=x, threshold=0.5)
barcode_assign_b2 <- merge(barcode_assign_b2, x[,c("Barcode","Assignment","Assignment_Probability")], by="Barcode")
barcode_assign_b2 <- merge(barcode_assign_b2, x_2[,c(2:4)], by="Barcode")
colnames(barcode_assign_b2)[6:9] <- c("ChimpRef_0.9","ChimpRef_0.9_AssignProb","ChimpRef_0.5","ChimpRef_0.5_AssignProb")

rm(x,x_2)

```

```{r}
cmo <- c("CMO301","CMO302","CMO303","CMO304","CMO305","CMO306","CMO307","CMO308","CMO309","CMO310","CMO311","CMO312")
```

```{r}

#batch 1 - human ortho-exon reference

#(1)load data
raw_data <- Read10X(data="./../scRNA/human_chimp_chondro_time_batch1_h/outs/multi/count/raw_feature_bc_matrix")

#(2)create seurat objects
data <- CreateSeuratObject(counts=raw_data$`Gene Expression`)

#(3)view details of seurat objects
data #19377 features, 2356630 samples

#(4)add percent of mitochondrial reads to metadata
mito <- c("MT-ATP6","MT-ATP8","MT-CO1","MT-CO2","MT-CO3","MT-CYB","MT-ND1","MT-ND2","MT-ND3","MT-ND4","MT-ND4L","MT-ND5")
data$percent.mt <- PercentageFeatureSet(data, features=mito)

#(5)add cellplex data to metadata
m <- match(rownames(data@meta.data),barcode_assign_b1$Barcode)
if(any(is.na(m))) cat("Not all barcodes are in data.\n")
data <- AddMetaData(data, barcode_assign_b1[m,]$HumanRef_0.5, col.name="CellPlex_HumanRef_0.5")
data <- AddMetaData(data, barcode_assign_b1[m,]$HumanRef_0.9, col.name="CellPlex_HumanRef_0.9")
data <- AddMetaData(data, barcode_assign_b1[m,]$HumanRef_0.5_AssignProb, col.name="CellPlex_HumanRef_0.5_AssignProb")
data <- AddMetaData(data, barcode_assign_b1[m,]$HumanRef_0.9_AssignProb, col.name="CellPlex_HumanRef_0.9_AssignProb")

#(6)plot QC metrics
table(data@meta.data$CellPlex_HumanRef_0.5)
table(data@meta.data$CellPlex_HumanRef_0.9)

VlnPlot(data,
        features=c("nFeature_RNA","nCount_RNA","percent.mt"),
        group.by="CellPlex_HumanRef_0.5",
        ncol=1,
        pt.size=0,
        #stack=TRUE
        )
mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.5 %in% cmo)])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.5=="CMO301")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.5=="CMO302")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.5=="CMO303")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.5=="CMO304")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.5=="CMO305")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.5=="CMO306")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.5=="CMO307")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.5=="CMO308")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.5=="CMO309")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.5=="CMO310")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.5=="CMO311")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.5=="CMO312")])

VlnPlot(data,
        features=c("nFeature_RNA","nCount_RNA","percent.mt"),
        group.by="CellPlex_HumanRef_0.9",
        ncol=1,
        pt.size=0,
        #stack=TRUE
        )
mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.9 %in% cmo)])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.9=="CMO301")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.9=="CMO302")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.9=="CMO303")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.9=="CMO304")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.9=="CMO305")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.9=="CMO306")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.9=="CMO307")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.9=="CMO308")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.9=="CMO309")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.9=="CMO310")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.9=="CMO311")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.9=="CMO312")])

rm(data,raw_data,m,mito)

```

```{r}

#batch 1 - chimp ortho-exon reference

#(1)load data
raw_data <- Read10X(data="./../scRNA/human_chimp_chondro_time_batch1_c/outs/multi/count/raw_feature_bc_matrix")

#(2)create seurat objects
data <- CreateSeuratObject(counts=raw_data$`Gene Expression`)

#(3)view details of seurat objects
data #19377 features, 2356630 samples

#(4)add percent of mitochondrial reads to metadata
mito <- c("MT-ATP6","MT-ATP8","MT-CO1","MT-CO2","MT-CO3","MT-CYB","MT-ND1","MT-ND2","MT-ND3","MT-ND4","MT-ND4L","MT-ND5")
data$percent.mt <- PercentageFeatureSet(data, features=mito)

#(5)add cellplex data to metadata
m <- match(rownames(data@meta.data),barcode_assign_b1$Barcode)
if(any(is.na(m))) cat("Not all barcodes are in data.\n")
data <- AddMetaData(data, barcode_assign_b1[m,]$ChimpRef_0.5, col.name="CellPlex_ChimpRef_0.5")
data <- AddMetaData(data, barcode_assign_b1[m,]$ChimpRef_0.9, col.name="CellPlex_ChimpRef_0.9")
data <- AddMetaData(data, barcode_assign_b1[m,]$ChimpRef_0.5_AssignProb, col.name="CellPlex_ChimpRef_0.5_AssignProb")
data <- AddMetaData(data, barcode_assign_b1[m,]$ChimpRef_0.9_AssignProb, col.name="CellPlex_ChimpRef_0.9_AssignProb")

#(6)plot QC metrics
table(data@meta.data$CellPlex_ChimpRef_0.5)
table(data@meta.data$CellPlex_ChimpRef_0.9)

VlnPlot(data,
        features=c("nFeature_RNA","nCount_RNA","percent.mt"),
        group.by="CellPlex_ChimpRef_0.5",
        ncol=1,
        pt.size=0,
        #stack=TRUE
        )
mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.5 %in% cmo[1:6])])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.5=="CMO301")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.5=="CMO302")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.5=="CMO303")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.5=="CMO304")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.5=="CMO305")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.5=="CMO306")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.5=="CMO307")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.5=="CMO308")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.5=="CMO309")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.5=="CMO310")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.5=="CMO311")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.5=="CMO312")])

VlnPlot(data,
        features=c("nFeature_RNA","nCount_RNA","percent.mt"),
        group.by="CellPlex_ChimpRef_0.9",
        ncol=1,
        pt.size=0,
        #stack=TRUE
        )
mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.9 %in% cmo[1:6])])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.9=="CMO301")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.9=="CMO302")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.9=="CMO303")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.9=="CMO304")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.9=="CMO305")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.9=="CMO306")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.9=="CMO307")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.9=="CMO308")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.9=="CMO309")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.9=="CMO310")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.9=="CMO311")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.9=="CMO312")])

rm(data,raw_data,m,mito)

```

```{r}

#batch 2 - human ortho-exon reference

#(1)load data
raw_data <- Read10X(data="./../scRNA/human_chimp_chondro_time_batch2_h/outs/multi/count/raw_feature_bc_matrix")

#(2)create seurat objects
data <- CreateSeuratObject(counts=raw_data$`Gene Expression`)

#(3)view details of seurat objects
data #19377 features, 1689884 samples

#(4)add percent of mitochondrial reads to metadata
mito <- c("MT-ATP6","MT-ATP8","MT-CO1","MT-CO2","MT-CO3","MT-CYB","MT-ND1","MT-ND2","MT-ND3","MT-ND4","MT-ND4L","MT-ND5")
data$percent.mt <- PercentageFeatureSet(data, features=mito)

#(5)add cellplex data to metadata
m <- match(rownames(data@meta.data),barcode_assign_b2$Barcode)
if(any(is.na(m))) cat("Not all barcodes are in data.\n")
data <- AddMetaData(data, barcode_assign_b2[m,]$HumanRef_0.5, col.name="CellPlex_HumanRef_0.5")
data <- AddMetaData(data, barcode_assign_b2[m,]$HumanRef_0.9, col.name="CellPlex_HumanRef_0.9")
data <- AddMetaData(data, barcode_assign_b2[m,]$HumanRef_0.5_AssignProb, col.name="CellPlex_HumanRef_0.5_AssignProb")
data <- AddMetaData(data, barcode_assign_b2[m,]$HumanRef_0.9_AssignProb, col.name="CellPlex_HumanRef_0.9_AssignProb")

#(6)plot QC metrics
table(data@meta.data$CellPlex_HumanRef_0.5)
table(data@meta.data$CellPlex_HumanRef_0.9)

VlnPlot(data,
        features=c("nFeature_RNA","nCount_RNA","percent.mt"),
        group.by="CellPlex_HumanRef_0.5",
        ncol=1,
        pt.size=0,
        #stack=TRUE
        )
mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.5 %in% cmo[1:6])])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.5=="CMO301")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.5=="CMO302")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.5=="CMO303")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.5=="CMO304")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.5=="CMO305")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.5=="CMO306")])

VlnPlot(data,
        features=c("nFeature_RNA","nCount_RNA","percent.mt"),
        group.by="CellPlex_HumanRef_0.9",
        ncol=1,
        pt.size=0,
        #stack=TRUE
        )
mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.9 %in% cmo[1:6])])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.9=="CMO301")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.9=="CMO302")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.9=="CMO303")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.9=="CMO304")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.9=="CMO305")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_HumanRef_0.9=="CMO306")])

rm(data,raw_data,m,mito)

```

```{r}

#batch 2 - chimp ortho-exon reference

#(1)load data
raw_data <- Read10X(data="./../scRNA/human_chimp_chondro_time_batch2_c/outs/multi/count/raw_feature_bc_matrix")

#(2)create seurat objects
data <- CreateSeuratObject(counts=raw_data$`Gene Expression`)

#(3)view details of seurat objects
data #19377 features, 1689884 samples

#(4)add percent of mitochondrial reads to metadata
mito <- c("MT-ATP6","MT-ATP8","MT-CO1","MT-CO2","MT-CO3","MT-CYB","MT-ND1","MT-ND2","MT-ND3","MT-ND4","MT-ND4L","MT-ND5")
data$percent.mt <- PercentageFeatureSet(data, features=mito)

#(5)add cellplex data to metadata
m <- match(rownames(data@meta.data),barcode_assign_b2$Barcode)
if(any(is.na(m))) cat("Not all barcodes are in data.\n")
data <- AddMetaData(data, barcode_assign_b2[m,]$ChimpRef_0.5, col.name="CellPlex_ChimpRef_0.5")
data <- AddMetaData(data, barcode_assign_b2[m,]$ChimpRef_0.9, col.name="CellPlex_ChimpRef_0.9")
data <- AddMetaData(data, barcode_assign_b2[m,]$ChimpRef_0.5_AssignProb, col.name="CellPlex_ChimpRef_0.5_AssignProb")
data <- AddMetaData(data, barcode_assign_b2[m,]$ChimpRef_0.9_AssignProb, col.name="CellPlex_ChimpRef_0.9_AssignProb")

#(6)plot QC metrics
table(data@meta.data$CellPlex_ChimpRef_0.5)
table(data@meta.data$CellPlex_ChimpRef_0.9)

VlnPlot(data,
        features=c("nFeature_RNA","nCount_RNA","percent.mt"),
        group.by="CellPlex_ChimpRef_0.5",
        ncol=1,
        pt.size=0,
        #stack=TRUE
        )
mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.5 %in% cmo[1:6])])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.5=="CMO301")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.5=="CMO302")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.5=="CMO303")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.5=="CMO304")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.5=="CMO305")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.5=="CMO306")])

VlnPlot(data,
        features=c("nFeature_RNA","nCount_RNA","percent.mt"),
        group.by="CellPlex_ChimpRef_0.9",
        ncol=1,
        pt.size=0,
        #stack=TRUE
        )
mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.9 %in% cmo[1:6])])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.9=="CMO301")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.9=="CMO302")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.9=="CMO303")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.9=="CMO304")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.9=="CMO305")])
#mean(data@meta.data$nCount_RNA[which(data@meta.data$CellPlex_ChimpRef_0.9=="CMO306")])

rm(data,raw_data,m,mito)

```

###UPDATE THIS SECTION
View 10X data metrics directly from Cellranger output files

###CHECK
```{r load libraries, message=FALSE}
#library(Seurat)
#library(dplyr)
#library(stringi)
#library(stringr)
#library(ggplot2)
#library(colorspace)
#library(RColorBrewer)
```

```{r compile summary metrics, results="hide"}

#Load batch info
#NEED TO CREATE THIS FILE
#batch <- read.csv(file='./data/scrna-batch.csv', header=TRUE, sep=",")
Sample_Name_at_Core <- c("H1_C1_d00","H1_C1_d0","H1_C1_m_d7","H1_C1_m_d14","H1_C1_c_d7","H1_C1_c_d14",
                         "H2_C4_d00","H2_C4_d0","H2_C4_m_d7","H2_C4_m_d14","H2_C4_c_d7","H2_C4_c_d14")

#Define methods of alignment - human (h) or chimp (c)
align <- c("h","c")

#FIX FROM HERE
#Load metrics data from full data

#Define directory containing data from one collection
dir_data <- paste0('./../scRNA/human_chimp_chondro_time_batch1_',align[1],'/outs/per_sample_outs/',Sample_Name_at_Core[1])
#Load summary metrics
temp <- read.csv(file=paste0(dir_data,'/metrics_summary.csv'))


metrics <- data.frame()
for (j in align) {
  i <- 1
  while (i <= length(batch$Sample_Name_at_Core)) {
    print(i)
    #Define directory containing data from one collection
    dir_data <- paste0('./../scRNA/human_chimp_chondro_time_batch1_',j,'/outs/per_sample_outs/',Sample_Name_at_Core[i])
    #Load summary metrics
    temp <- read.csv(file=paste0(dir_data,'/metrics_summary.csv'))
    #Process metrics
    temp[,c(1:4,19:20)] <- lapply(temp[,c(1:4,19:20)], function(x) as.numeric(gsub(",","",as.character(x))))
    temp[,c(5:18)] <- lapply(temp[,c(5:18)], function(x) as.numeric(gsub("%","",as.character(x)))/100)
    temp$Collection.Name <- batch$Sample_Name_at_Core[i]
    temp$Alignment.Type <- j
    #Add data to dataframe
    metrics <- rbind(metrics,temp)
    i <- i+1
  }
}
rm(dir_data,temp,i,j)
#Fix metrics details so that plotting is easier
metrics$Collection.Name <- factor(metrics$Collection.Name,
                                  levels=c("GHO-1","GHO-2","GHO-4",
                                           "HOU-10","HOU-12","HOU-16",
                                           "HOU-1","HOU-2","HOU-4",
                                           "HOU-9","HOU-11","HOU-15",
                                           "HOU-5","HOU-6","HOU-8",
                                           "HOU-17","HOU-19","HOU-21",
                                           "HOU-18","HOU-20","HOU-22"))
metrics$reads_prop <- metrics$Number.of.Reads / metrics$Number.of.Reads
metrics$reads_mapped_to_genome <- metrics$Number.of.Reads * metrics$Reads.Mapped.to.Genome
metrics$reads_mapped_confidently_to_genome <- metrics$Number.of.Reads * metrics$Reads.Mapped.Confidently.to.Genome
metrics$reads_mapped_confidently_to_exonic_regions <- metrics$Number.of.Reads * metrics$Reads.Mapped.Confidently.to.Exonic.Regions
metrics$reads_mapped_confidently_to_transcriptome <- metrics$Number.of.Reads * metrics$Reads.Mapped.Confidently.to.Transcriptome
```

## Visualize read alignment metrics:

```{r}
#total number of reads
ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'), aes(x=Collection.Name, y=Number.of.Reads)) +
  geom_col(position="dodge") +
  theme_classic() +
  theme(axis.text.x=element_text(size=6)) +
  labs(title="Number of Reads Sequenced",y="Number of Reads",x="Collection") +
  ylim(0,300000000)
```

```{r}
#percent sequence saturation
ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'), aes(x=Collection.Name, y=Sequencing.Saturation)) +
  geom_col(position="dodge") +
  theme_classic() +
  theme(axis.text.x=element_text(size=6)) +
  labs(title="Sequencing Saturation",y="Sequencing Saturation",x="Collection") +
  ylim(0,1)
```

```{r}
#mean reads per cell
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'), aes(x=Collection.Name, y=Mean.Reads.per.Cell)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Reads per Cell (hg38)",y="Mean Reads per Cell",x="Collection") +
                           ylim(0,110000)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2'), aes(x=Collection.Name, y=Mean.Reads.per.Cell)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Reads per Cell (panTro6)",y="Mean Reads per Cell",x="Collection") +
                           ylim(0,110000))), ncol=1)
```

```{r}
#median genes per cell
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'), aes(x=Collection.Name, y=Median.Genes.per.Cell)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Genes per Cell (ortho-exon hg38)",y="Median Genes per Cell",x="Collection") +
                           ylim(0,6500)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2'), aes(x=Collection.Name, y=Median.Genes.per.Cell)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Genes per Cell (ortho-exon panTro6)",y="Median Genes per Cell",x="Collection") +
                           ylim(0,6500))), ncol=1)
```

```{r}
#median UMI per cell
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'), aes(x=Collection.Name, y=Median.UMI.Counts.per.Cell)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="UMIs per Cell (ortho-exon hg38)",y="Median UMI Counts per Cell",x="Collection") +
                           ylim(0,31000)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2'), aes(x=Collection.Name, y=Median.UMI.Counts.per.Cell)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="UMIs per Cell (ortho-exon panTro6)",y="Median UMI Counts per Cell",x="Collection") +
                           ylim(0,31000))), ncol=1)
```

```{r}
#proportion of Q30 bases in barcode, rna read, and umi
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'), aes(x=Collection.Name, y=Q30.Bases.in.Barcode)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Proportion Q30: Barcode (ortho-exon hg38)",y="Q30 Bases in Barcode",x="Collection") +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'), aes(x=Collection.Name, y=Q30.Bases.in.UMI)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Proportion Q30: UMI (ortho-exon hg38)",y="Q30 Bases in UMI",x="Collection") +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'), aes(x=Collection.Name, y=Q30.Bases.in.RNA.Read)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Proportion Q30: RNA (ortho-exon hg38)",y="Q30 Bases in RNA Read",x="Collection") +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2'), aes(x=Collection.Name, y=Q30.Bases.in.Barcode)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Proportion Q30: Barcode (ortho-exon panTro6)",y="Q30 Bases in Barcode",x="Collection") +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2'), aes(x=Collection.Name, y=Q30.Bases.in.UMI)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Proportion Q30: UMI (ortho-exon panTro6)",y="Q30 Bases in UMI",x="Collection") +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2'), aes(x=Collection.Name, y=Q30.Bases.in.RNA.Read)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Proportion Q30: RNA (ortho-exon panTro6)",y="Q30 Bases in RNA Read",x="Collection") +
                           ylim(0,1))), ncol=3)
```

```{r}
#reads mapped to genome, confidently mapped to genome, confidently mapped to exons, and confidently mapped to transcriptome
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2')) +
                           geom_col(aes(x=Collection.Name, y=reads_prop), position="dodge", alpha=0.2) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.to.Genome), position="dodge", alpha=0.3) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Genome), position="dodge", alpha=0.4) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Exonic.Regions), position="dodge", alpha=0.6) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Transcriptome), position="dodge", alpha=1) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Proportion of Mapped Reads (ortho-exon hg38)",y="Mapped Reads",x="Collection") +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2')) +
                           geom_col(aes(x=Collection.Name, y=reads_prop), position="dodge", alpha=0.2) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.to.Genome), position="dodge", alpha=0.3) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Genome), position="dodge", alpha=0.4) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Exonic.Regions), position="dodge", alpha=0.6) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Transcriptome), position="dodge", alpha=1) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Proportion of Mapped Reads (ortho-exon panTro6)",y="Mapped Reads",x="Collection") +
                           ylim(0,1))), ncol=1)
```

```{r}
#reads mapped to genome, confidently mapped to genome, confidently mapped to exons, and confidently mapped to transcriptome
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2')) +
                           geom_col(aes(x=Collection.Name, y=Number.of.Reads), position="dodge", alpha=0.2) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_to_genome), position="dodge", alpha=0.3) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_genome), position="dodge", alpha=0.4) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_exonic_regions), position="dodge", alpha=0.6) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_transcriptome), position="dodge", alpha=1) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Numbers of Mapped Reads (ortho-exon hg38)",y="Mapped Reads",x="Collection") +
                           ylim(0,300000000)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2')) +
                           geom_col(aes(x=Collection.Name, y=Number.of.Reads), position="dodge", alpha=0.2) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_to_genome), position="dodge", alpha=0.3) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_genome), position="dodge", alpha=0.4) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_exonic_regions), position="dodge", alpha=0.6) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_transcriptome), position="dodge", alpha=1) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Numbers of Mapped Reads (ortho-exon panTro6)",y="Mapped Reads",x="Collection") +
                           ylim(0,300000000))), ncol=1)
```

```{r}
#reads mapped to intergenic regions, intronic regions, and antisense to gene
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'),
                                aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Intergenic.Regions)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Reads Mapped to Intergenic Regions (ortho-exon hg38)",
                                y="Reads Mapped to Intergenic Regions",
                                x="Collection") +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'),
                                aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Intronic.Regions)) +
                            geom_col(position="dodge") +
                            theme_classic() +
                            theme(axis.text.x=element_text(size=6)) +
                            labs(title="Reads Mapped to Intronic Regions (ortho-exon hg38)",
                                 y="Reads Mapped to Intronic Regions",
                                 x="Collection") +
                            ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'),
                                aes(x=Collection.Name, y=Reads.Mapped.Antisense.to.Gene)) +
                            geom_col(position="dodge") +
                            theme_classic() +
                            theme(axis.text.x=element_text(size=6)) +
                            labs(title="Reads Mapped to Intergenic Regions (ortho-exon hg38)",
                                 y="Reads Mapped to Intergenic Regions",
                                 x="Collection") +
                            ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2'),
                                aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Intergenic.Regions)) +
                            geom_col(position="dodge") +
                            theme_classic() +
                            theme(axis.text.x=element_text(size=6)) +
                            labs(title="Reads Mapped to Intergenic Regions (ortho-exon panTro6)",
                                 y="Reads Mapped to Intergenic Regions",
                                 x="Collection") +
                            ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2'),
                                aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Intronic.Regions)) +
                            geom_col(position="dodge") +
                            theme_classic() +
                            theme(axis.text.x=element_text(size=6)) +
                            labs(title="Reads Mapped to Intronic Regions (ortho-exon panTro6)",
                                 y="Reads Mapped to Intronic Regions",
                                 x="Collection") +
                            ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2'),
                                aes(x=Collection.Name, y=Reads.Mapped.Antisense.to.Gene)) +
                            geom_col(position="dodge") +
                            theme_classic() +
                            theme(axis.text.x=element_text(size=6)) +
                            labs(title="Reads Mapped to Intergenic Regions (ortho-exon panTro6)",
                                 y="Reads Mapped to Intergenic Regions",
                                 x="Collection") +
                            ylim(0,1))), ncol=3)
```

```{r}
#fraction of reads in cells
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'), aes(x=Collection.Name, y=Fraction.Reads.in.Cells)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Fraction of Reads in Cells (ortho-exon hg38)",y="Fraction of Reads in Cells",x="Collection") +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2'), aes(x=Collection.Name, y=Fraction.Reads.in.Cells)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Fraction of Reads in Cells (ortho-exon panTro6)",y="Fraction of Reads in Cells",x="Collection") +
                           ylim(0,1))), ncol=1)
```

```{r}
#Compare alignment proportions
mean(subset(metrics,Alignment.Type=='HumanOrthoV2')$Reads.Mapped.Confidently.to.Transcriptome) #65.17%
mean(subset(metrics,Alignment.Type=='ChimpOrthoV2')$Reads.Mapped.Confidently.to.Transcriptome) #63.41%
#Summarize metrics in plots for figures
metrics.h <- subset(metrics,Alignment.Type=='HumanOrthoV2')
metrics.h$Collection <- c("H1-I/C1-I","H1-M/C1-M","H1-O/C1-O",
                          "H1-r2-I/C1-r2-I","H1-r2-M/C1-r2-M","H1-r2-O/C1-r2-O",
                          "H2-I/C2-I","H2-M/C2-M","H2-O/C2-O",
                          "H3-I/C3-I","H3-M/C3-M","H3-O/C3-O",
                          "H4-I/C4-I","H4-M/C4-M","H4-O/C4-O",
                          "H5-I/C5-I","H5-M/C5-M","H5-O/C5-O",
                          "H6-I/C6-I","H6-M/C6-M","H6-O/C6-O")
metrics.h$Collection <- factor(metrics.h$Collection,
                               levels=c("H1-I/C1-I","H1-M/C1-M","H1-O/C1-O",
                                        "H1-r2-I/C1-r2-I","H1-r2-M/C1-r2-M","H1-r2-O/C1-r2-O",
                                        "H2-I/C2-I","H2-M/C2-M","H2-O/C2-O",
                                        "H3-I/C3-I","H3-M/C3-M","H3-O/C3-O",
                                        "H4-I/C4-I","H4-M/C4-M","H4-O/C4-O",
                                        "H5-I/C5-I","H5-M/C5-M","H5-O/C5-O",
                                        "H6-I/C6-I","H6-M/C6-M","H6-O/C6-O"))
metrics.c <- subset(metrics,Alignment.Type=='ChimpOrthoV2')
metrics.c$Collection <- c("H1-I/C1-I","H1-M/C1-M","H1-O/C1-O",
                          "H1-r2-I/C1-r2-I","H1-r2-M/C1-r2-M","H1-r2-O/C1-r2-O",
                          "H2-I/C2-I","H2-M/C2-M","H2-O/C2-O",
                          "H3-I/C3-I","H3-M/C3-M","H3-O/C3-O",
                          "H4-I/C4-I","H4-M/C4-M","H4-O/C4-O",
                          "H5-I/C5-I","H5-M/C5-M","H5-O/C5-O",
                          "H6-I/C6-I","H6-M/C6-M","H6-O/C6-O")
metrics.c$Collection <- factor(metrics.c$Collection,
                               levels=c("H1-I/C1-I","H1-M/C1-M","H1-O/C1-O",
                                        "H1-r2-I/C1-r2-I","H1-r2-M/C1-r2-M","H1-r2-O/C1-r2-O",
                                        "H2-I/C2-I","H2-M/C2-M","H2-O/C2-O",
                                        "H3-I/C3-I","H3-M/C3-M","H3-O/C3-O",
                                        "H4-I/C4-I","H4-M/C4-M","H4-O/C4-O",
                                        "H5-I/C5-I","H5-M/C5-M","H5-O/C5-O",
                                        "H6-I/C6-I","H6-M/C6-M","H6-O/C6-O"))
```

```{r}
ggplot(metrics.h) +
  geom_col(aes(x=Collection.Name, y=Number.of.Reads), position="dodge", alpha=0.2) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_to_genome), position="dodge", alpha=0.3) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_genome), position="dodge", alpha=0.4) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_exonic_regions), position="dodge", alpha=0.5) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_transcriptome), position="dodge", alpha=1) +
  theme_classic() +
  theme(axis.text.x=element_text(size=12), axis.text.y=element_text(size=12)) +
  labs(title="Numbers of Mapped Reads (Genome: hg38, Annotation: ortho-exon)",y="Mapped Reads",x="Collection") +
  ylim(0,300000000) +
  scale_x_discrete(labels=c("GHO-1"="H1+C1\nTime 0",
                            "GHO-2"="H1+C1\nTime 1",
                            "GHO-4"="H1+C1\nTime 2",
                            "HOU-10"="H1r2+C1r2\nTime 0",
                            "HOU-12"="H1r2+C1r2\nTime 1",
                            "HOU-16"="H1r2+C1r2\nTime 2",
                            "HOU-1"="H2+C2\nTime 0",
                            "HOU-2"="H2+C2\nTime 1",
                            "HOU-4"="H2+C2\nTime 2",
                            "HOU-9"="H3+C3\nTime 0",
                            "HOU-11"="H3+C3\nTime 1",
                            "HOU-15"="H3+C3\nTime 2",
                            "HOU-5"="H4+C4\nTime 0",
                            "HOU-6"="H4+C4\nTime 1",
                            "HOU-8"="H4+C4\nTime 2",
                            "HOU-17"="H5+C5\nTime 0",
                            "HOU-19"="H5+C5\nTime 1",
                            "HOU-21"="H5+C5\nTime 2",
                            "HOU-18"="H6+C6\nTime 0",
                            "HOU-20"="H6+C6\nTime 1",
                            "HOU-22"="H6+C6\nTime 2"))
```

```{r}
ggplot(metrics.c) +
  geom_col(aes(x=Collection.Name, y=Number.of.Reads), position="dodge", alpha=0.2) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_to_genome), position="dodge", alpha=0.3) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_genome), position="dodge", alpha=0.4) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_exonic_regions), position="dodge", alpha=0.5) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_transcriptome), position="dodge", alpha=1) +
  theme_classic() +
  theme(axis.text.x=element_text(size=12), axis.text.y=element_text(size=12)) +
  labs(title="Numbers of Mapped Reads (Genome: panTro6, Annotation: ortho-exon)",y="Mapped Reads",x="Collection") +
  ylim(0,300000000) +
  scale_x_discrete(labels=c("GHO-1"="H1+C1\nTime 0",
                            "GHO-2"="H1+C1\nTime 1",
                            "GHO-4"="H1+C1\nTime 2",
                            "HOU-10"="H1r2+C1r2\nTime 0",
                            "HOU-12"="H1r2+C1r2\nTime 1",
                            "HOU-16"="H1r2+C1r2\nTime 2",
                            "HOU-1"="H2+C2\nTime 0",
                            "HOU-2"="H2+C2\nTime 1",
                            "HOU-4"="H2+C2\nTime 2",
                            "HOU-9"="H3+C3\nTime 0",
                            "HOU-11"="H3+C3\nTime 1",
                            "HOU-15"="H3+C3\nTime 2",
                            "HOU-5"="H4+C4\nTime 0",
                            "HOU-6"="H4+C4\nTime 1",
                            "HOU-8"="H4+C4\nTime 2",
                            "HOU-17"="H5+C5\nTime 0",
                            "HOU-19"="H5+C5\nTime 1",
                            "HOU-21"="H5+C5\nTime 2",
                            "HOU-18"="H6+C6\nTime 0",
                            "HOU-20"="H6+C6\nTime 1",
                            "HOU-22"="H6+C6\nTime 2"))
```

```{r}
ggplot(metrics.h) +
  geom_col(aes(x=Collection.Name, y=Number.of.Reads), position="dodge", alpha=0.2) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_to_genome), position="dodge", alpha=0.3) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_exonic_regions), position="dodge", alpha=0.4) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_transcriptome), position="dodge", alpha=1) +
  theme_classic() +
  theme(axis.text.x=element_text(size=12), axis.text.y=element_text(size=12)) +
  labs(title="Numbers of Mapped Reads (Genome: hg38, Annotation: ortho-exon)",y="Mapped Reads",x="Collection") +
  ylim(0,300000000) +
  scale_x_discrete(labels=c("GHO-1"="H1+C1\nTime 0",
                            "GHO-2"="H1+C1\nTime 1",
                            "GHO-4"="H1+C1\nTime 2",
                            "HOU-10"="H1r2+C1r2\nTime 0",
                            "HOU-12"="H1r2+C1r2\nTime 1",
                            "HOU-16"="H1r2+C1r2\nTime 2",
                            "HOU-1"="H2+C2\nTime 0",
                            "HOU-2"="H2+C2\nTime 1",
                            "HOU-4"="H2+C2\nTime 2",
                            "HOU-9"="H3+C3\nTime 0",
                            "HOU-11"="H3+C3\nTime 1",
                            "HOU-15"="H3+C3\nTime 2",
                            "HOU-5"="H4+C4\nTime 0",
                            "HOU-6"="H4+C4\nTime 1",
                            "HOU-8"="H4+C4\nTime 2",
                            "HOU-17"="H5+C5\nTime 0",
                            "HOU-19"="H5+C5\nTime 1",
                            "HOU-21"="H5+C5\nTime 2",
                            "HOU-18"="H6+C6\nTime 0",
                            "HOU-20"="H6+C6\nTime 1",
                            "HOU-22"="H6+C6\nTime 2"))
```

```{r}
metrics$Stage <- factor(rep(c("Time 0","Time 1","Time 2"),7),
                        levels=c("Time 0","Time 1","Time 2"))
ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2')) +
                                geom_col(aes(x=Stage, y=Number.of.Reads), position="dodge", alpha=0.2) +
                                geom_col(aes(x=Stage, y=reads_mapped_to_genome), position="dodge", alpha=0.3) +
                                geom_col(aes(x=Stage, y=reads_mapped_confidently_to_exonic_regions), position="dodge", alpha=0.6) +
                                geom_col(aes(x=Stage, y=reads_mapped_confidently_to_transcriptome), position="dodge", alpha=1) +
                                theme(axis.text.x=element_text(size=6)) +
                                labs(title="Numbers of Mapped Reads (ortho-exon hg38)",y="Mapped Reads",x="Collection")
```

## Conclusions

* Number of total reads sequenced varied slightly across collections
* Number of reads that were retained (confidently mapped to the transcriptome) were similar across collections
* Number of reads that were retained (confidently mapped to the transcriptome) were similar whether using hg38 or panTro6 reference
* 10X collections from Time 2 generally had higher sequencing saturation than from Time 0 or Time 1
* Collection HOU-15 (H3+C3 Time 2) has higher reads/cell, UMIs/cell, and genes/per cell
